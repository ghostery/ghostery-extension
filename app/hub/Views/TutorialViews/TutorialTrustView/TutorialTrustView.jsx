/**
 * Tutorial Trust View Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import PropTypes from 'prop-types';

/**
 * A Functional React component for rendering the Tutorial Trust and Restrict View
 * @return {JSX} JSX for rendering the Tutorial Trust and Restrict View of the Hub app
 * @memberof HubComponents
 */
const TutorialTrustView = ({ isAndroid }) => (
	<div className="TutorialTrustView TutorialView--mediumFlexColumn row align-center-middle">
		<div className="columns small-10 small-offset-1 medium-8 large-6">
			<div className="TutorialView__imageTitle">
				{t('simple_view')}
			</div>
			<img
				className="TutorialTrustView__image trustrestrict-simple"
				src={`/app/images/hub/tutorial/trustrestrict-simple${isAndroid ? '-android' : ''}.png`}
				alt={t('simple_view')}
			/>
			{ !isAndroid && (
				<div>
					<div className="TutorialView__imageTitle">
						{t('detailed_view')}
					</div>
					<img
						className="TutorialTrustView__image trustrestrict-detailed"
						src="/app/images/hub/tutorial/trustrestrict-detailed.png"
						alt={t('detailed_view')}
					/>
				</div>
			)}
		</div>
		<div className="columns small-12 medium-8 large-4">
			<div className="TutorialView__title">
				{t('hub_tutorial_trust_title')}
			</div>
			<div className="TutorialView__tagline">
				{t('hub_tutorial_trust_tagline')}
			</div>
			<div className="TutorialTrustView__key">
				<div className="TutorialView__keyButton trust button secondary hollow">
					{t('summary_trust_site')}
				</div>
				<div className="TutorialView__keyText">
					{t('hub_tutorial_trust_trust_site')}
				</div>
				<div className="TutorialView__keyButton restrict button secondary hollow">
					{t('summary_restrict_site')}
				</div>
				<div className="TutorialView__keyText TutorialView--removeMarginBottom">
					{t('hub_tutorial_trust_restrict_site')}
				</div>
			</div>
		</div>
	</div>
);

TutorialTrustView.propTypes = {
	isAndroid: PropTypes.bool.isRequired
};

export default TutorialTrustView;
