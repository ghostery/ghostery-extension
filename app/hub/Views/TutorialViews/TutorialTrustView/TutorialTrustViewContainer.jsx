/**
 * Tutorial Trust View Container
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';

import TutorialTrustView from './TutorialTrustView';

/**
 * @class Implement the Tutorial Trust View for the Ghostery Hub
 * @extends Component
 * @memberof HubComponents
 */
class TutorialTrustViewContainer extends Component {
	constructor(props) {
		super(props);

		// TODO call setTutorialNavigation action
		const { index } = this.props;
		this.props.actions.setTutorialNavigation({
			activeIndex: index,
			hrefPrev: `/tutorial/${index - 1}`,
			hrefNext: `/tutorial/${index + 1}`,
			hrefDone: '/',
			textPrev: t('hub_setup_nav_previous'),
			textNext: t('hub_setup_nav_next'),
			textDone: t('hub_setup_exit_flow'),
		});
	}

	render() {
		return <TutorialTrustView />;
	}
}

// PropTypes ensure we pass required props of the correct type
TutorialTrustViewContainer.propTypes = {
	index: PropTypes.number.isRequired,
	actions: PropTypes.shape({
		setTutorialNavigation: PropTypes.func.isRequired,
	}).isRequired,
};

export default TutorialTrustViewContainer;
