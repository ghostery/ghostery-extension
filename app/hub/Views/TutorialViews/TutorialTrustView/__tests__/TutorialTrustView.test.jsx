/**
 * Tutorial Trust View Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TutorialTrustView from '../TutorialTrustView';

describe('app/hub/Views/TutorialViews/TutorialTrustView component', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('tutorial trust view is rendered correctly', () => {
			const component = renderer.create(<TutorialTrustView isAndroid={false} />).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('the happy path of the component', () => {
			const component = shallow(<TutorialTrustView isAndroid={false} />);
			expect(component.find('.TutorialTrustView').length).toBe(1);
			expect(component.find('.TutorialTrustView__image').length).toBe(2);
			expect(component.find('.TutorialTrustView__key').length).toBe(1);
		});
	});
});
