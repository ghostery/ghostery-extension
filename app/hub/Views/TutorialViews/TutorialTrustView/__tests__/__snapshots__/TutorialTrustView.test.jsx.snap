// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/hub/Views/TutorialViews/TutorialTrustView component Snapshot tests with react-test-renderer tutorial trust view is rendered correctly 1`] = `
<div
  className="TutorialTrustView TutorialView--mediumFlexColumn row align-center-middle"
>
  <div
    className="columns small-10 small-offset-1 medium-8 large-6"
  >
    <div
      className="TutorialView__imageTitle"
    >
      simple_view
    </div>
    <img
      alt="simple_view"
      className="TutorialTrustView__image trustrestrict-simple"
      src="/app/images/hub/tutorial/trustrestrict-simple.png"
    />
    <div>
      <div
        className="TutorialView__imageTitle"
      >
        detailed_view
      </div>
      <img
        alt="detailed_view"
        className="TutorialTrustView__image trustrestrict-detailed"
        src="/app/images/hub/tutorial/trustrestrict-detailed.png"
      />
    </div>
  </div>
  <div
    className="columns small-12 medium-8 large-4"
  >
    <div
      className="TutorialView__title"
    >
      hub_tutorial_trust_title
    </div>
    <div
      className="TutorialView__tagline"
    >
      hub_tutorial_trust_tagline
    </div>
    <div
      className="TutorialTrustView__key"
    >
      <div
        className="TutorialView__keyButton trust button secondary hollow"
      >
        summary_trust_site
      </div>
      <div
        className="TutorialView__keyText"
      >
        hub_tutorial_trust_trust_site
      </div>
      <div
        className="TutorialView__keyButton restrict button secondary hollow"
      >
        summary_restrict_site
      </div>
      <div
        className="TutorialView__keyText TutorialView--removeMarginBottom"
      >
        hub_tutorial_trust_restrict_site
      </div>
    </div>
  </div>
</div>
`;
